### PR TITLE
sci-biology/fsl: version bump

### PR DESCRIPTION
**Pull request by TheChymera**
Package-Manager: Portage-2.3.3, Repoman-2.3.1

reiteration of #718

### CODE REVIEW

**Comment by TheChymera** [comment]
@jlec @kiwifb?

File: sci-biology/fsl/fsl-5.0.9.ebuild [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+		-i $(grep -rl "\$FSLDIR/etc" src/*)
+
+	sed -e "s:\$FSLDIR/doc:${EPREFIX}/usr/share/fsl/doc:g" \
+		-e "s:\${FSLDIR}/doc:${EPREFIX}/usr/share/fsl/doc:g" \
+		-i $(grep -rl "\$FSLDIR/doc" src/*)
+
+	sed -i "s:\'\${FSLDIR}\'/doc:${EPREFIX}/usr/share/fsl/doc:g" \
+		$(grep -rl "\'\${FSLDIR}\'/doc" src/*)
+
+	sed -i -e "s:\$FSLDIR/etc:/etc:g" `grep -rlI \$FSLDIR/etc *`
```
Comment (kiwifb) [inline sci-biology/fsl/fsl-5.0.9.ebuild]: You probably want to `${EPREFIX}` in front of `/etc` in the replacement part of sed. Also keeping the `-rlI` you could probably unit this sed line with the one acting on `$(grep -rl "\$FSLDIR/etc" src/*)` surely `$(grep -rlI "\$FSLDIR/etc" *)` will cover everything you need here.

File: sci-biology/fsl/fsl-5.0.9.ebuild [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	sed -e "s:\$FSLDIR/data:${EPREFIX}/usr/share/fsl/data:g" \
+		-e "s:\${FSLDIR}/data:${EPREFIX}/usr/share/fsl/data:g" \
+		-i $(grep -rl "\$FSLDIR/data" src/*)
+
+	sed -e "s:\$FSLDIR/etc:${EPREFIX}/etc:g" \
+		-e "s:\${FSLDIR}/etc:${EPREFIX}/etc:g" \
+		-i $(grep -rl "\$FSLDIR/etc" src/*)
+
+	sed -e "s:\$FSLDIR/doc:${EPREFIX}/usr/share/fsl/doc:g" \
+		-e "s:\${FSLDIR}/doc:${EPREFIX}/usr/share/fsl/doc:g" \
```
Comment (kiwifb) [inline sci-biology/fsl/fsl-5.0.9.ebuild]: That line is probably only working by chance on some files. This isn't what we look for in grep. And after close examination this is not what we look for in the next sed line either.

**Comment by kiwifb** [comment]
I should have some look at it again to see if we can out of that `sed` forest on to something easier. A patch would be huge and you'd probably want a script to generate it. Part of the problem is that upstream doesn't have a consistent coding style - amongst other things :open_mouth:

**Comment by TheChymera** [comment]
@kiwifb you can pull from my repo and we can both push to this PR.

**Comment by TheChymera** [comment]
@jlec @kiwifb  this is very strange. For some reason, this now fails to apply one of the patches:

```
zenbookhost ~ # emerge -av =fsl-5.0.9

These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild     U  ] sci-biology/fsl-5.0.9::neurogentoo [5.0.8::science] 0 KiB

Total: 1 package (1 upgrade), Size of downloads: 0 KiB

Would you like to merge these packages? [Yes/No] y

>>> Verifying ebuild manifests

>>> Emerging (1 of 1) sci-biology/fsl-5.0.9::neurogentoo
 * fsl-5.0.9-sources.tar.gz SHA256 SHA512 WHIRLPOOL size ;-) ...            [ ok ]
>>> Unpacking source...
>>> Unpacking fsl-5.0.9-sources.tar.gz to /var/tmp/portage/sci-biology/fsl-5.0.9/work
>>> Source unpacked in /var/tmp/portage/sci-biology/fsl-5.0.9/work
>>> Preparing source in /var/tmp/portage/sci-biology/fsl-5.0.9/work/fsl ...
 * Applying fsl-5.0.9-setup.patch ...                                       [ ok ]
 * Applying fsl-5.0.9-headers.patch ...
The text leading up to this was:
--------------------------
|--- src/miscvis/writepng.c.orig	2013-10-19 00:18:32.415203592 +1300
|+++ src/miscvis/writepng.c	2013-10-19 00:19:05.414255132 +1300
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored                                                     [ !! ]
 * ERROR: sci-biology/fsl-5.0.9::neurogentoo failed (prepare phase):
 *   patch -p1  failed with /home/chymera/src/neurogentoo/sci-biology/fsl/files/fsl-5.0.9-headers.patch
 * 
 * Call stack:
 *               ebuild.sh, line  115:  Called src_prepare
 *             environment, line 1025:  Called default
 *      phase-functions.sh, line  807:  Called default_src_prepare
 *      phase-functions.sh, line  872:  Called __eapi6_src_prepare
 *             environment, line  178:  Called eapply '/home/chymera/src/neurogentoo/sci-biology/fsl/files/fsl-5.0.9-setup.patch' '/home/chymera/src/neurogentoo/sci-biology/fsl/files/fsl-5.0.9-headers.patch' '/home/chymera/src/neurogentoo/sci-biology/fsl/files/fsl-5.0.8-fsldir_redux.patch'
 *             environment, line  337:  Called _eapply_patch '/home/chymera/src/neurogentoo/sci-biology/fsl/files/fsl-5.0.9-headers.patch'
 *             environment, line  275:  Called __helpers_die 'patch -p1  failed with /home/chymera/src/neurogentoo/sci-biology/fsl/files/fsl-5.0.9-headers.patch'
 *   isolated-functions.sh, line  117:  Called die
 * The specific snippet of code:
 *   		die "$@"
 * 
 * If you need support, post the output of `emerge --info '=sci-biology/fsl-5.0.9::neurogentoo'`,
 * the complete build log and the output of `emerge -pqv '=sci-biology/fsl-5.0.9::neurogentoo'`.
 * The complete build log is located at '/var/tmp/portage/sci-biology/fsl-5.0.9/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/sci-biology/fsl-5.0.9/temp/environment'.
 * Working directory: '/var/tmp/portage/sci-biology/fsl-5.0.9/work/fsl'
 * S: '/var/tmp/portage/sci-biology/fsl-5.0.9/work/fsl'

>>> Failed to emerge sci-biology/fsl-5.0.9, Log file:

>>>  '/var/tmp/portage/sci-biology/fsl-5.0.9/temp/build.log'

 * Messages for package sci-biology/fsl-5.0.9:

 * ERROR: sci-biology/fsl-5.0.9::neurogentoo failed (prepare phase):
 *   patch -p1  failed with /home/chymera/src/neurogentoo/sci-biology/fsl/files/fsl-5.0.9-headers.patch
 * 
 * Call stack:
 *               ebuild.sh, line  115:  Called src_prepare
 *             environment, line 1025:  Called default
 *      phase-functions.sh, line  807:  Called default_src_prepare
 *      phase-functions.sh, line  872:  Called __eapi6_src_prepare
 *             environment, line  178:  Called eapply '/home/chymera/src/neurogentoo/sci-biology/fsl/files/fsl-5.0.9-setup.patch' '/home/chymera/src/neurogentoo/sci-biology/fsl/files/fsl-5.0.9-headers.patch' '/home/chymera/src/neurogentoo/sci-biology/fsl/files/fsl-5.0.8-fsldir_redux.patch'
 *             environment, line  337:  Called _eapply_patch '/home/chymera/src/neurogentoo/sci-biology/fsl/files/fsl-5.0.9-headers.patch'
 *             environment, line  275:  Called __helpers_die 'patch -p1  failed with /home/chymera/src/neurogentoo/sci-biology/fsl/files/fsl-5.0.9-headers.patch'
 *   isolated-functions.sh, line  117:  Called die
 * The specific snippet of code:
 *   		die "$@"
 * 
 * If you need support, post the output of `emerge --info '=sci-biology/fsl-5.0.9::neurogentoo'`,
 * the complete build log and the output of `emerge -pqv '=sci-biology/fsl-5.0.9::neurogentoo'`.
 * The complete build log is located at '/var/tmp/portage/sci-biology/fsl-5.0.9/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/sci-biology/fsl-5.0.9/temp/environment'.
 * Working directory: '/var/tmp/portage/sci-biology/fsl-5.0.9/work/fsl'
 * S: '/var/tmp/portage/sci-biology/fsl-5.0.9/work/fsl'
 * 
 * The following package has failed to build, install, or execute postinst:
 * 
 *  (sci-biology/fsl-5.0.9:0/0::neurogentoo, ebuild scheduled for merge), Log file:
 *   '/var/tmp/portage/sci-biology/fsl-5.0.9/temp/build.log'
 * 
```

I can't find anything wrong with it. Can you help me figure out what's wrong here?

**Comment by kiwifb** [comment]
I am fairly sure that's because you switched to EAPI 6 method of patching. `epatch` was trying variation of the patch command from `patch -p0` (which is what would work for your failing patch) to `patch -p9`. In EAPI 6, `eapply` and the `default` method use `patch -p1` exclusively. So you would have to add a directory level up, say from `src/` to `fsl/src/` in fsl-5.0.9-headers.patch.

You may want to check the redux patch for the sam eproblem.

**Comment by TheChymera** [comment]
@kiwifb thanks :) @jlec This should be working now

**Comment by TheChymera** [comment]
@kiwifb @jlec  can we pull this now?

File: sci-biology/fsl/fsl-5.0.9.ebuild [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+DEPEND="${COMMON_DEPEND}"
+RDEPEND="${COMMON_DEPEND}
+	dev-lang/tcl:0=
+	dev-lang/tk:0=
+	"
+
+S=${WORKDIR}/${PN}
+
+PATCHES=(
+		"${FILESDIR}/${PN}"-5.0.9-setup.patch \
```
Comment (jlec) [inline sci-biology/fsl/fsl-5.0.9.ebuild]: Please remove 1 of the leading tabs and the trailing `\`

**Pull request closed without merging**